src/components/Router/StepComponentMap.js

```diff
@@ -10,15 +10,16 @@ import CrossDeviceLink from '../crossDevice/CrossDeviceLink'
 import ClientSuccess from '../crossDevice/ClientSuccess'
 
 export const componentsList = ({flow, documentType, steps, mobileFlow}) => {
-  const hasComplete = steps[steps.length -1].type === 'complete'
-  const captureSteps = mobileFlow ? clientCaptureSteps(steps, hasComplete) : steps
+  const captureSteps = mobileFlow ? clientCaptureSteps(steps) : steps
   return flow === 'captureSteps' ?
     createComponentList(captureStepsComponents(documentType, mobileFlow), captureSteps) :
-    createComponentList(crossDeviceComponents, crossDeviceSteps(hasComplete))
+    createComponentList(crossDeviceComponents, crossDeviceSteps(steps))
 }
 
-const clientCaptureSteps = (steps, hasComplete) =>
-  hasComplete ? steps : [...steps, {type: 'complete'}]
+const hasCompleteStep = (steps) => steps.some(step => step.type === 'complete')
```

Comment:
Minor: `step => step.type === 'complete'` can be made into a constant `isComplete` and reused